master/internal/api_runs.go

```diff
@@ -239,3 +251,201 @@ func sortRuns(sortString *string, runQuery *bun.SelectQuery) error {
 	}
 	return nil
 }
+
+func (a *apiServer) MoveRuns(
+	ctx context.Context, req *apiv1.MoveRunsRequest,
+) (*apiv1.MoveRunsResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// check that user can view source project
+	srcProject, err := a.GetProjectByID(ctx, req.SourceProjectId, *curUser)
```

Comment:
blocking: do we check that the user can delete the run?

MoveExperiments does the check here
https://github.com/determined-ai/determined/blob/b726cf9eb7e84e74068fb182e0b0a1693769ca04/master/internal/experiment/bulk_action.go#L654

Reply:
Added permission check below